vault/vault.go

```diff
@@ -193,6 +225,79 @@ func (v *vaultSecrets) ListSecrets() ([]string, error) {
 	return nil, secrets.ErrNotSupported
 }
 
+func (v *vaultSecrets) read(path string) (*api.Secret, error) {
+	secretValue, err := v.lockedRead(path)
+	if v.isTokeExpired(err) {
+		if err = v.renewToken(); err != nil {
+			return nil, err
```

Comment:
lets return the original error here instead of the error received as part of renewToken

Reply:
Updated.